dkist/net/attrs.py

```diff
@@ -278,3 +279,61 @@ def __init__(self, bottom_left, *, top_right=None, width: u.deg = None,
 
     def collides(self, other):
         return isinstance(other, self.__class__)
+
+# averageDatasetSpectralSamplingMin, averageDatasetSpectralSamplingMax
+class SpectralSampling(_Range):
+    """
+    The average spectral sampling of a dataset.
+
+    Parameters
+    ----------
+    spectralmin : `u.Quantity`
+        The minimum value of the average spectral sampling to search between.
+
+    spectralmax : `u.Quantity`
+        The maximum value of the average spectral sampling to search between.
+    """
+    u.quantity_input(equivalencies=u.spectral())
+    def __init__(self, spectralmin: u.nm, spectralmax: u.nm):
+        super().__init__(spectralmin, spectralmax)
+
+    def collides(self, other):
+        return isinstance(other, self.__class__)
+
+# averageDatasetSpatialSamplingMin, averageDatasetSpatialSamplingMax
+class SpatialSampling(_Range):
+    """
+    The average spatial sampling of a dataset.
+
+    Parameters
+    ----------
+    spatialmin :
+        The minimum value of the average spatial sampling to search between.
+
+    spatialmax :
+        The maximum value of the average spatial sampling to search between.
+    """
+    def __init__(self, spatialmin: u.arcsec/u.pix, spatialmax: u.arcsec/u.pix):
```

Comment:
add `u.quantity_input` here as well, but no equivalency is needed this time:
```suggestion
    @u.quantity_input
    def __init__(self, spatialmin: u.arcsec/u.pix, spatialmax: u.arcsec/u.pix):
```